File: 프로그래머스/2/42842. 카펫/카펫.js
// 블럭 개수의 총합 => 면적
// 가능한 가로 세로 길이를 확인해보면서,
// brown === 테두리 개수, yellow === 안에 있는 거 개수면 리턴
// 테두리 개수 = (column * 2) + (row - 2) * 2

function solution(brown, yellow) {
    const total = brown + yellow;
    for (let i = 1; i < total; i++) {
        if (total % i !== 0) continue;
        const [ row, column ] = [ i, total / i];
        const border = (column * 2) + (row - 2) * 2;
        if (border === brown) {
            return [ column, row ];
        }
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    size = brown + yellow
    for a in range(3, size // 2 + 1):
        if size % a == 0:
            b = size / a
            if (a - 2) * (b - 2) == yellow:
                return [ b, a ]
        
    
    return [ -1, -1 ]
    

# brown + yellow = n인 ab 중에 (a - 2) * (b - 2) = yellow인 것 찾기
# 48
# 48 * 1
# 24 * 2
# 12 * 4
# 6 * 8일 때 yellow 개수: (6 - 2) * (8 - 2) = 4 * 6 = 24

# 시간복잡도: O(n + m)
# 공간복잡도: O(1)


# 24
# = 24 * 1 => brown의 개수: (26 * 3) - yellow
# = 12 * 2
# = 8 * 3
# = 6 * 4 => brown의 개수: (8 * 6) - yellow

# 2 * 1일 때 brown의 개수: (4 * 3) - yellow

